### PR TITLE
clang-tidy report issues with Medium priority

### PR DESCRIPTION
**Pull request by varadarajkumar**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
clang-tidy medium priority issues fixed.

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by kitaisreal** [comment]
@varadarajkumar which warning these are ?

**Comment by varadarajkumar** [comment]
> @varadarajkumar which warning these are ?

@kitaisreal, these are clang-tidy report issues.

**Comment by kitaisreal** [comment]
@varadarajkumar could you please specify which issue type it is from this list https://clang.llvm.org/extra/clang-tidy/checks/list.html, for us to include in clang-tidy.

**Comment by kitaisreal** [comment]
@Mergifyio update

**Comment by varadarajkumar** [comment]
Hi @kitaisreal, 
Please find the report for reference.
image
image
image
image

**Comment by kitaisreal** [comment]
@Mergifyio update

**Comment by kitaisreal** [comment]
@Mergifyio update

**Review by kitaisreal — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
This pull request appeared to be wrong.
It also lacks of a proper description with explanation of the changes.

File: src/Storages/StorageDistributed.h [hunk 1]
```diff
@@ -43,7 +43,7 @@ class StorageDistributed final : public shared_ptr_helper<StorageDistributed>, p
     friend class StorageSystemDistributionQueue;
 
 public:
-    ~StorageDistributed() override;
+    virtual ~StorageDistributed() override;
```
Comment (alexey-milovidov) [inline src/Storages/StorageDistributed.h:46]: According to our style, if you write `override` you don't need to write `virtual`.

File: src/IO/ReadBufferFromFileDescriptor.h [hunk 2]
```diff
@@ -39,6 +39,10 @@ class ReadBufferFromFileDescriptor : public ReadBufferFromFileBase
     {
     }
 
+    virtual ~ReadBufferFromFileDescriptor() override
```
Comment (alexey-milovidov) [inline src/IO/ReadBufferFromFileDescriptor.h:42]: This code does not make sense.

**Pull request merged**